### PR TITLE
Simplify making parameterised script instances

### PR DESCRIPTION
**Pull request by sjoerdvisscher**
Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you changed any Haskell files:
    - [x] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by michaelpj** [comment]
I think originally I thought it would be good for people to understand the mechanics of what's going on here, but in practice I think they're almost exclusively just passing a single argument, so this seems fine!

File: plutus-playground-server/usecases/Crowdfunding.hs [hunk 1]
```diff
@@ -107,11 +107,11 @@ instance Scripts.ScriptType Crowdfunding where
     type instance DatumType Crowdfunding = PubKeyHash
 
 scriptInstance :: Campaign -> Scripts.ScriptInstance Crowdfunding
-scriptInstance cmp = Scripts.validator @Crowdfunding
-    ($$(PlutusTx.compile [|| mkValidator ||]) `PlutusTx.applyCode` PlutusTx.liftCode cmp)
+scriptInstance = Scripts.validatorParam @Crowdfunding
+    $$(PlutusTx.compile [|| mkValidator ||])
     $$(PlutusTx.compile [|| wrap ||])
```
Comment (michaelpj) [inline plutus-playground-server/usecases/Crowdfunding.hs:112]: Ugh, this is still so ugly. If you figure out a way to avoid it I'd be very happy!

Comment (sjoerdvisscher) [reply plutus-playground-server/usecases/Crowdfunding.hs:112]: Yeah, but at least it now looks the same in most examples, with the only difference the type in the type application and the choice between `validator` and `validatorParam`.

**Comment by sjoerdvisscher** [comment]
There are already enough mechanics to have to understand before a user can write a complete Plutus contract. 
And anyways it's good practice to create a datatype with all the parameters if you need more than one.

File: plutus-playground-server/usecases/Crowdfunding.hs (hunk 1 above)
Comment (michaelpj) [reply plutus-playground-server/usecases/Crowdfunding.hs:112]: Yeah, that wasn't a comment on this PR, I just was struck again by its ugliness :(

**Comment by michaelpj** [comment]
Hm. A bit weird that this changes the script hashes.

**Comment by sjoerdvisscher** [comment]
Leaving everything else the same, the script hash changes by just moving the part 
```hakell
`PlutusTx.applyCode` PlutusTx.liftCode param
```
to a different module. Does that make sense to you?

**Review by j-mueller — APPROVED** [review]
Nice. I suppose if we have several parameters we can just make a tuple.

**Comment by michaelpj** [comment]
I don't care enough about the hash change mystery to investigate right now, let's move on!

**Pull request merged**